packages/orchestrator/cmd/clean-nfs-cache/main.go

```diff
@@ -151,116 +158,103 @@ type results struct {
+		deletedBytes:     r.deletedBytes + other.deletedBytes,
+		lastAccessed:     append(r.lastAccessed, other.lastAccessed...),
+		createdDurations: append(r.createdDurations, other.createdDurations...),
+	}
+}
+
+func deleteOldestFiles(cache *pkg.ListingCache, files []pkg.File, opts opts, diskInfo *pkg.DiskInfo, areWeDone func() bool, deleteCount int64) (results, error) {
 	now := time.Now()
 	var results results
-	for _, file := range files {
+	for index, file := range files {
 		if opts.dryRun {
 			fmt.Printf("DRY RUN: would delete %q (%d bytes, created: %s, last access: %s)\n",
-				file.path, file.size,
-				time.Since(file.btime).Round(time.Second),
-				time.Since(file.atime).Round(time.Minute))
+				file.Path, file.Size,
+				time.Since(file.BTime).Round(time.Second),
+				time.Since(file.ATime).Round(time.Minute))
 		} else {
-			// remove file
-			fmt.Printf("deleting %q (%d bytes) ... ", file.path, file.size)
-			if err := os.Remove(file.path); err != nil {
-				// if we fail to delete the file, try the next one
-				fmt.Printf("failed to delete %q: %v\n", file.path, err)
+			// remove File
+			fmt.Printf("deleting #%d: %q (%d bytes) ... ", index+1, file.Path, file.Size)
+			if err := os.Remove(file.Path); err != nil {
+				// if we fail to delete the File, try the next one
+				fmt.Printf("failed to delete %q: %v\n", file.Path, err)
 				continue
 			}
 			fmt.Print("\n")
 		}
 
+		cache.Decache(filepath.Dir(file.Path))
 		results.deletedFiles++
-		results.deletedBytes += file.size
-		results.lastAccessed = append(results.lastAccessed, now.Sub(file.atime))
-		results.createdDurations = append(results.createdDurations, time.Since(file.btime))
+		results.deletedBytes += file.Size
+		results.lastAccessed = append(results.lastAccessed, now.Sub(file.ATime))
+		results.createdDurations = append(results.createdDurations, time.Since(file.BTime))
 
-		// record the file as free space
-		diskInfo.used -= file.size
-		if areWeDone() {
+		// record the File as free space
+		diskInfo.Used -= file.Size
+		if areWeDone() || results.deletedFiles >= deleteCount {
 			// we're done!
 			return results, nil
 		}
 	}
 
 	return results, fmt.Errorf("%w: target: %.2f%% < actual: %.2f%%",
 		ErrFail, opts.targetDiskUsagePercent,
-		(float64(diskInfo.used)/float64(diskInfo.total))*100)
+		(float64(diskInfo.Used)/float64(diskInfo.Total))*100)
 }
 
-type diskInfo struct {
-	total, used int64
+func sortFilesByATime(files []pkg.File) {
+	sort.Slice(files, func(i, j int) bool {
+		return files[j].ATime.After(files[i].ATime)
+	})
 }
 
-func getDiskInfo(ctx context.Context, path string) (diskInfo, error) {
-	// Execute: df <path>
-	cmd := exec.CommandContext(ctx, "df", path)
-	out, err := cmd.CombinedOutput()
-	if err != nil {
-		return diskInfo{}, fmt.Errorf("df command failed: %w: %s", err, strings.TrimSpace(string(out)))
+func reportGetFilesProgress(usedFiles, dupeHits int) {
+	total := usedFiles + dupeHits
+	if total > 0 && total%100 != 0 {
```

Comment:
i think this should be 
```suggestion
	if total > 0 && total%100 == 0 {
```